[SH-77] 방 리스트 조회 기능 버그 수정

leewontae0 opened the pull request:
## 📌 관련 이슈
- Jira: [SH-77]

## 📝 PR 요약
ReactiveRedisTemplate 빈 등록 시 발생하던 에러 수정

## ⚒️ 주요 변경 사항
<!-- 구체적인 변경 내용을 bullet point로 나열해주세요 -->
- RedisConfig.java
- RedisConfig.java 파일을 통해 ReactiveRedisTemplate type마다 bean 생성

- SecurityConfig.java
- 방 리스트 조회 기능 테스트 중에, SecurityConfig.java 파일 때문에 401 에러 발생
- "/lobby" 추가

- Room.java
- deserialize 할 때, 기본 생성자 없을 시 오류 발생
- 기본 생성자 추가

## 🧪 테스트 체크리스트
- [ 단위 테스트 ] 

## ✅ 체크리스트
- [x] 로컬에서 테스트를 완료했나요?
- [x] Assignees를 등록했나요?
- [x] Label을 지정했나요?

[SH-77]: https://snowhite.atlassian.net/browse/SH-77?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

leewontae0 committed:
fix : ReactiveRedisTemplate Bean 등록

leewontae0 committed:
fix : RedisConfig 수정

leewontae0 committed:
fix : 방 리스트 조회 기능 버그 수정

- ReactiveRedisTemplate 빈 등록 시 발생하던 에러 해결

leewontae0 committed:
Merge branch 'develop' into SH-77

leewontae0 committed:
style : LobbyService Qualifier annotation 수정

leewontae0 committed:
Merge branch 'SH-77' of
https://github.com/snowhite-hyu/backend into SH-77

lheesoo1020 commented on src/main/java/com/snowhite/server/config/RedisConfig.java:
```diff
@@ -0,0 +1,48 @@
+        return new ReactiveRedisTemplate<>(factory, context);
+    }
+
+    @Bean(name="reactiveRedisTemplateForRooms")
+    public ReactiveRedisTemplate<Long, Room> reactiveRedisTemplateForRooms(
+            ReactiveRedisConnectionFactory factory) {
+        RedisSerializationContext<Long, Room> context = RedisSerializationContext
+                .<Long, Room>newSerializationContext(new GenericToStringSerializer<>(Long.class))
+                .value(new Jackson2JsonRedisSerializer<>(Room.class))
+                .build();
+        return new ReactiveRedisTemplate<>(factory, context);
+    }
+}
```
room의 경우 키값을 "room:123"과 같이 하기로 했으니 해당 redisTemplate의 key를 String으로 바꿔줘야 할 것 같아요

leewontae0 replied:
수정했습니다.
<String, Long> 타입의 redisTemplate 은 삭제하고,
말씀해주신 패턴 매칭을 도입해 <Long, Room> redisTemplate만 남겨뒀습니다.

leewontae0 committed:
fix : JwtWebFilter, SecurityConfig 컴파일 에러 수정

- JwtWebFilter에서 findById 함수의 인자로 String을 줌으로써
- 발생하는 에러 수정
- SecurityConfig , 대신 . 사용된 에러 수정

leewontae0 committed:
refactor : <String, Long> redis 삭제

- 기존에 roomId set을 별도로 redis에 저장했었으나,
- 패턴 매칭 기능을 통해 존재의의가 사라짐으로 인해 삭제

lheesoo1020 reviewed (approved):
확인입니다~~~

Ryuga2967 reviewed (approved):
머지하고 Redis 관련해서 수정 되는지 확인하고 Revert 시키면 될 것 같습니다.
